<launch>
    <param name="use_sim_time" value="false"/>
    <param name="do_acc_boards" type="bool" value="false"/>
    <param name="end_process" type="bool" value="false"/>
    <param name="pause_processd" type="bool" value="false"/>
    <param name="cam_paused" type="bool" value="false"/>
    <param name="livox_paused" type="bool" value="false"/>
    
    <!-- the number of frames accumulated to do the calibration, default:60 -->
    <arg name="ns_l" default="laser"/>
    <arg name="ns_c" default="camera"/>
    <arg name="center_laser_tp" default="/$(arg ns_l)/centers_cloud"/>
    <arg name="center_cam_tp" default="/$(arg ns_c)/cam_pattern/centers_cloud"/>
    <arg name="center_cam2d_tp" default="/$(arg ns_c)/cam_pattern/cam_2d_circle_center"/>
    <arg name="max_frame" default="60"/>
    <arg name="feature_file_name" default="features_info_$(arg ns_l)_to_$(arg ns_c)"/>


    <param name="max_frame" type="int" value="$(arg max_frame)"/>

    <!-- Main Node -->
    <node pkg="lvt2calib" type="pattern_collection_lc" name="pattern_collection_lc" output="screen">
    
        <remap from="~cloud_laser" to="$(arg center_laser_tp)"/>
        <remap from="~cloud_cam" to="$(arg center_cam_tp)"/>
        <remap from="~cloud_cam2d" to="$(arg center_cam2d_tp)"/>

        <param name="ns_lv" value="$(arg ns_l)"/>
        <param name="ns_cv" value="$(arg ns_c)"/>
        <!-- use centroid of laser patterns? default:true -->
        <param name="useCentroid_laser" value="true" />

        <!-- Save file? -->        
        <param name="save_final_data" value="true"/>
        <!-- MUST TRUE!!! -->

        <!-- File Path -->
        <param name="result_dir_" type="string" value="$(find lvt2calib)/data/pattern_collection_result/"/>
        <param name="feature_file_name" type="string" value="$(arg feature_file_name)"/>
    
    </node>
</launch>